egglog-bridge/src/lib.rs

```diff
@@ -259,19 +306,63 @@ impl EGraph {
         }
     }
 
+    fn record_term_consistency(
+        state: &mut ExecutionState,
+        table: TableId,
+        ts_counter: CounterId,
+        from: Value,
+        to: Value,
+    ) {
+        if from == to {
+            return;
+        }
+        let ts = Value::from_usize(state.read_counter(ts_counter));
+        state.stage_insert(table, &[from, to, ts]);
+    }
+
+    fn canonicalize_term_id(&mut self, term_id: Value) -> Value {
+        let table = self.db.get_table(self.term_consistency_table);
+        table
+            .get_row(&[term_id])
+            .map(|row| row.vals[1])
+            .unwrap_or(term_id)
+    }
+
     fn term_table(&mut self, table: TableId) -> TableId {
         let spec = self.db.get_table(table).spec();
         match self.term_tables.entry(spec.n_keys) {
             Entry::Occupied(o) => *o.get(),
             Entry::Vacant(v) => {
+                let term_index = spec.n_keys + 1;
+                let term_consistency_table = self.term_consistency_table;
```

Comment:
Nit: unnecessary extra variable- I would just refer to self.term_consistency_table everywhere

Reply:
We're binding it here because we `move` it in the closure further down and want to avoid moving self along with it. We're basically using this to "build the closure by hand" like in C++, but without the nice capture group syntax.

Reply:
Ohh makes sense